tableconvert/src/tableconvert.c

```diff
@@ -338,6 +351,31 @@ void cb_table_convert_type(G_GNUC_UNUSED GtkMenuItem *menuitem, G_GNUC_UNUSED gp
 	convert_to_table(TRUE, GPOINTER_TO_INT(gdata));
 }
 
+void cb_table_convert_change_document(G_GNUC_UNUSED GtkMenuItem *menuitem, G_GNUC_UNUSED gpointer gdata)
+{
+	set_activate_state();
+}
+
+void set_activate_state()
+{
+	// getting document
+	GeanyDocument *doc = NULL;
+	doc = document_get_current();
+
+	if (
+		doc != NULL && (
```

Comment:
This should also check if there is a current selection or not. The function ```convert_to_table()``` is only doing something if there is a selection (```if (sci_has_selection(doc->editor->sci)) ...```).

Reply:
Good idea -- but not yet sure it's  really needed.

Reply:
Well, the menu item is doing nothing if there is no selection. IMHO it should check the selection or if not the user should get a message that a selection is needed before selecting the menu item.

Reply:
It's the wrong place for getting a selection

Reply:
Can you explain what you mean please? I mean it would be good to check if there is a selection, not really getting it (there is the ```doc``` pointer and that's all we need?).

Reply:
At this point yes. It's either the menu is active or not. The actually handling i done in convert_to_table().

Reply:
Ok, when the events are fired, a document is most likely just opened or switched to so there most likely cannot be a selection (yet).